### PR TITLE
feat(core): add ability to search clusters by labels

### PR DESCRIPTION
**Pull request by maggieneterval**
Adding label search functionality as a separate PR from label filter functionality as the latter is a bit messier (#3238)

a0bd8221-3a10-4d85-a1b1-49c2da51989d

### CODE REVIEW

**Comment by lwander** [comment]
The demo is curious, why does applying a more restrictive filter (more labels) result in more results? Are the labels being OR'd instead of AND'd?

**Comment by maggieneterval** [comment]
@lwander Yeah I tried to keep it consistent with how `clusters` search works, which is an OR, but if you think AND makes more sense for labels I can switch it and just clarify that in the help box. (https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/cluster/filter/ClusterFilterService.spec.ts#L135)

**Comment by lwander** [comment]
Ah that's difficult... I'd prefer to remain consistent with what people expect coming from kubernetes, especially since this label query will help them build expressions to use either in pipelines or manifests when selecting groups of resources in k8s, via spinnaker.

I wonder if we can guide users towards this feature by expanding it from being a custom search query to something with a little more treatment in a future PR

**Comment by maggieneterval** [comment]
Makes sense, updated search to match on AND.
Incoming PR for a hopefully more intuitive UI like this:

e1rapwuvaot

However, sounds like we don't need to have multiple acceptable values for a given label key so I'll change that multi-select to a single select.

**Comment by lwander** [comment]
That looks awesome

File: app/scripts/modules/core/src/cluster/filter/ClusterFilterService.ts [hunk 1]
```diff
@@ -334,6 +334,25 @@ export class ClusterFilterService {
...
+        let labelKey: string;
+        let labelValue: string;
+        if (label.includes('=')) {
+          [labelKey, labelValue] = label.split('=');
+        }
+        if (!labelKey || !labelValue) {
+          return false;
+        }
+        return some(serverGroup.labels || {}, (val: string, key: string) => {
+          return labelKey.toLowerCase() === key.toLowerCase() && val.toLowerCase().includes(labelValue.toLowerCase());
```
Comment (lwander) [inline app/scripts/modules/core/src/cluster/filter/ClusterFilterService.ts]: I think these are case-sensitive... do the k8s docs say otherwise?

Comment (maggieneterval) [reply app/scripts/modules/core/src/cluster/filter/ClusterFilterService.ts]: Ah, another case of inconsistency with existing search filters vs. inconsistency with k8s patterns -- I'll remove case insensitivity + partial value match acceptance

**Pull request merged**